### ISSUE TITLE
cant watch shows and a small question

### ISSUE THREAD

**Issue by papi028** [stivengjinaj/StiFlix#4]
so i cant watch shows here specifically "breaking bad"   it just sends you to vidcloud website with an error
buh also wanted to ask how you get the video source is only embed or u can actually get the source url and use it in a custom player if you want if yes can u show me how it can be done

i actaully have a movie web my own buh i need a movie and shows provider buh i dont want an embed, i want the links so i can use them in my player

**Comment by stivengjinaj** [comment]
1) I just checked breaking bad. I noticed that the servers do not have the movie. Usually there are 4 servers where you can watch the shows. Omega server is the last one and I noticed that it has the show but Omega is the only server that usually has ads. However, given that the first 3 servers do not have the show, it shows the error message. If you hover on the top left part of the window, the dropdown to change servers will appear. Thanks for point this out, cause I also noticed some UI errors on mobile version. I'll try to add a check for 404 response code when the movie is not found in order not to display the server. Unfortunately, I can't control which movie the site can display, cause I depend on the servers and as you saw the movies are embedded, so if the movie is not in the server, I can't do much.

2) As per the links question, I tried to find a way to get the actual files and not be limited to embedd links, but I couldn't. My initial goal was to make a custom video player similar to netflix's. If you need links, check server implementation in the repository. Also Braflix movie site has its own video player and I guess they are getting movies from the same links as mines, but I wasn't able to understand how they do it. My guess is that they are directly linked to the providers and have special access that we don't have. However you can try and reverse engineer braflix website by yourself and if you find anything you can contact me again.

**Comment by papi028** [comment]
> 1. I just checked breaking bad. I noticed that the servers do not have the movie. Usually there are 4 servers where you can watch the shows. Omega server is the last one and I noticed that it has the show but Omega is the only server that usually has ads. However, given that the first 3 servers do not have the show, it shows the error message. If you hover on the top left part of the window, the dropdown to change servers will appear. Thanks for point this out, cause I also noticed some UI errors on mobile version. I'll try to add a check for 404 response code when the movie is not found in order not to display the server. Unfortunately, I can't control which movie the site can display, cause I depend on the servers and as you saw the movies are embedded, so if the movie is not in the server, I can't do much.
> 
>     2. As per the links question, I tried to find a way to get the actual files and not be limited to embedd links, but I couldn't. My initial goal was to make a custom video player similar to netflix's. If you need links, check server implementation in the repository. Also Braflix movie site has its own video player and I guess they are getting movies from the same links as mines, but I wasn't able to understand how they do it. My guess is that they are directly linked to the providers and have special access that we don't have. However you can try and reverse engineer braflix website by yourself and if you find anything you can contact me again.


yea braflix seems to be getting the  actual links from rabbitstream it really does seem they have exclusive access or maybe they scrapping , i can't  try to reverse engineer their website because u cant even view their network logs , just will detects it and send you to google.com so thats not possible their alternative site i saw actaully doesnt get the links buh uses an embed so thats a dead end, i was actaully using consumet's flixhq provider buh it recently died so that why im tryna find other providers there are other embed that actually have good player ui (https://embed.su) buh i just want something i can control there's a lot of feature i have to implement in the player and using an embed wont help

**Comment by stivengjinaj** [comment]
Yes I understand. I would like to implement my features too. If you want to inspect Braflix traffic, try Requestly cause it doesn't require you to open console but it will intercept all the requests. Another problem is that the sites that display the actual file will never display the full file. I researched a bit about this and it turns out the file is splitted in pieces. Check this italian streaming site: streamingcommunity.computer . It has its own video player.

**Comment by papi028** [comment]
oh i see i kinda looks like an embed with the jwplayer buh dont know how they customized the player maybe just made the controls on top buh im still tryna find a provider probably an one which has an api and maybe accepts tmdb id if it doesnt support tmdb i might have to use matching system buh im still searching  on here, im gonna try the requestly on several sites and see what i can find
i will let you know when  i find something
